iroh-resolver/src/unixfs_builder.rs

```diff
@@ -102,92 +117,67 @@ impl Directory {
             let node = UnixfsNode::Directory(Node { outer, inner });
             yield node.encode()?;
         }
+        .boxed_local()
     }
 }
 
-/// Constructs a UnixFS file.
-pub struct FileBuilder {
-    name: Option<String>,
-    content: Option<Pin<Box<dyn AsyncRead>>>,
-    chunker: Chunker,
+enum Content {
+    Reader(Pin<Box<dyn AsyncRead>>),
+    Path(PathBuf),
 }
 
-impl Debug for FileBuilder {
+impl Debug for Content {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let content = if self.content.is_some() {
-            "Some(Box<AsyncRead>)"
-        } else {
-            "None"
-        };
+        match self {
+            Content::Reader(_) => write!(f, "Content::Reader(Pin<Box<dyn AsyncRead>>)"),
+            Content::Path(p) => write!(f, "Content::Path({})", p.display()),
+        }
+    }
+}
 
-        f.debug_struct("FileBuilder")
-            .field("name", &self.name)
-            .field("content", &content)
-            .field("chunker", &self.chunker)
-            .finish()
+impl PartialEq for Content {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Content::Reader(_), Content::Reader(_)) => true,
```

Comment:
this should not return `true`, as this means that two different content readers will be considered equal

Reply:
Ahhh, thank you, I meant to ask you about this!!! 

Wasn't sure if the readers should always return `false`, or if there was some way to figure out how to force checking for an equivalency.

Reply:
just return false, we can't really compare them